### PR TITLE
Small improvements in setup.sh

### PR DESCRIPTION
**Pull request by simonsmh**
Support more system.
A user-friendly way to use virtualenv.
There is no need to reinstall python packages when update any more.

https://github.com/PokemonGoF/PokemonGo-Bot/issues/3796
https://github.com/PokemonGoF/PokemonGo-Bot/pull/3702
https://github.com/PokemonGoF/PokemonGo-Bot/pull/3662
#3820

### CODE REVIEW

**Comment by simonsmh** [comment]
@TheSavior

**Comment by simonsmh** [comment]
@solderzzc merge plz

**Comment by TheSavior** [comment]
:+1:

File: setup.sh [hunk 1]
```diff
@@ -8,17 +8,18 @@ cd $pokebotpath
 git pull
 git submodule update --init --recursive
 git submodule foreach git pull origin master
-virtualenv .
 source bin/activate
+pip install -r requirements.txt --upgrade
```
Comment (nivong) [inline setup.sh]: Please dubble check @thesavior this is now dubble please remove this.

File: setup.sh [hunk 2]
```diff
@@ -56,27 +57,42 @@ echo "Edit ./configs/config.json to modify any other config."
...
-elif [ -f /etc/redhat-release ]
+sudo apt-get -y install python python-pip python-dev gcc make git
+elif [ -x "$(command -v yum)" ]
 then
 echo "You are on CentOS/RedHat"
-sudo yum -y install epel-release
-sudo yum -y install python-pip
-elif [ "$(uname -s)" == "Darwin" ]
+sudo yum -y install epel-release gcc make
+sudo yum -y install python-pip python-devel
```
Comment (nivong) [inline setup.sh]: Is it for sure python-Devel and not python-dev?

Comment (HongxuChen) [inline setup.sh]: @nivong after a quick search, "python-devel" should be correct: http://stackoverflow.com/questions/23541205/cant-install-python-dev-on-centos-6-5

File: setup.sh (hunk 1 above)
Comment (simonsmh) [inline setup.sh]: no its not

File: setup.sh (hunk 2 above)
Comment (nivong) [inline setup.sh]: ah you are right :)

**Comment by nivong** [comment]
would it be also a good idea to fix this?
https://github.com/PokemonGoF/PokemonGo-Bot/issues/3855

**Comment by simonsmh** [comment]
@nivong Python 2.7.3 is too old. Now even Debian 8 have 2.7.9 support and this won't be fixed. The users should either upgrade their system or upgrade python only.

**Comment by simonsmh** [comment]
Where is greenlet?
Not in requirements.txt?

**Comment by simonsmh** [comment]
To fix it
root@simonsmh:~# cat /etc/issue
Debian GNU/Linux 8 \n \l

It should be 7 here.
And Debian 7 doesn't support apt, only apt-get. 
I'd prefer python version detection but I don't know how.

File: setup.sh [hunk 3]
```diff
@@ -56,27 +57,42 @@ echo "Edit ./configs/config.json to modify any other config."
...
+elif [ -x "$(command -v pacman)" ]
 then
-echo "You are on Mac os"
```
Comment (gsmith257) [inline setup.sh]: why would you remove Mac OS from here?

Comment (simonsmh) [inline setup.sh]: l26

Comment (gsmith257) [inline setup.sh]: What does that mean. (Not super experienced with shell)

Comment (simonsmh) [inline setup.sh]: Line 62..

Comment (gsmith257) [inline setup.sh]: Oh ok thanks

**Pull request merged**